trace_decoder/src/compact/compact_to_partial_trie.rs

```diff
@@ -8,183 +11,261 @@ use mpt_trie::{
 };
 
 use super::compact_prestate_processing::{
-    AccountNodeCode, AccountNodeData, CompactParsingResult, LeafNodeData, NodeEntry, WitnessEntry,
+    AccountNodeCode, AccountNodeData, CompactParsingError, CompactParsingResult, LeafNodeData,
+    NodeEntry, WitnessEntry,
 };
 use crate::{
-    types::{CodeHash, HashedAccountAddr, TrieRootHash, EMPTY_CODE_HASH, EMPTY_TRIE_HASH},
+    decoding::TrieType,
+    types::{
+        CodeHash, HashedAccountAddr, HashedAccountAddrNibbles, TrieRootHash, EMPTY_CODE_HASH,
+        EMPTY_TRIE_HASH,
+    },
     utils::{h_addr_nibs_to_h256, hash},
 };
 
-#[derive(Debug, Default)]
-pub(super) struct CompactToPartialTrieExtractionOutput {
-    pub(super) trie: HashedPartialTrie,
+pub trait CompactToPartialTrieExtractionOutput {
+    fn process_branch(
+        &mut self,
+        curr_key: Nibbles,
+        children: &[Option<Box<NodeEntry>>],
+    ) -> CompactParsingResult<()> {
+        for (i, slot) in children.iter().enumerate().take(16) {
+            if let Some(child) = slot {
+                // TODO: Seriously update `mpt_trie` to have a better API...
+                let mut new_k = curr_key;
+                new_k.push_nibble_back(i as Nibble);
+                create_partial_trie_from_compact_node_rec(new_k, child, self)?;
+            }
+        }
 
-    // TODO: `code` is ever only available for storage tries, so we should come up with a better
-    // API that represents this...
-    pub(super) code: HashMap<CodeHash, Vec<u8>>,
-}
+        Ok(())
+    }
 
-pub(super) fn create_partial_trie_from_remaining_witness_elem(
-    remaining_entry: WitnessEntry,
-) -> CompactParsingResult<CompactToPartialTrieExtractionOutput> {
-    let remaining_node = remaining_entry
-        .into_node()
-        .expect("Final node in compact entries was not a node! This is a bug!");
+    fn process_code(&mut self, c_bytes: Vec<u8>) -> CompactParsingResult<()>;
```

Comment:
Some doc on what the methods are doing for this new trait would be nice

Reply:
Yeah fair I agree. Let me add this.